tests/test_postgres_logical_replication.py

```diff
@@ -555,10 +637,137 @@ def test_run(self):
 
         lsn_3 = bookmark['lsn']
         self.assertTrue(lsn_3 >= lsn_2)
+        #----------------------------------------------------------------------
+        # invoke the sync job again after deleting a record using the 'id IN (SELECT ...)' format
+        #----------------------------------------------------------------------
+        print("delete row from source db")
+        with db_utils.get_test_connection('dev') as conn:
+            with conn.cursor(cursor_factory=psycopg2.extras.DictCursor) as cur:
+                cur.execute("DELETE FROM {} WHERE id IN (SELECT id FROM {} WHERE id=2)".format(canonicalized_table_name(test_schema_name, test_table_name, cur), canonicalized_table_name(test_schema_name, test_table_name, cur)))
+
+        sync_job_name = runner.run_sync_mode(self, conn_id)
+
+        # verify tap and target exit codes
+        exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
+        menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
+
+        record_count_by_stream = runner.examine_target_output_file(self,
+                                                                   conn_id,
+                                                                   self.expected_sync_streams(),
+                                                                   self.expected_pks())
+
+
+        self.assertEqual(record_count_by_stream, { 'postgres_logical_replication_test': 2 })
+        records_by_stream = runner.get_records_from_target_output()
+
+        for stream, recs in records_by_stream.items():
+            # verify the persisted schema was correct
+            self.assertEqual(recs['schema'],
```

Comment:
Should this be `self.assertDictEqual`?

Reply:
Maybe it can be, it seems like `self.assertEqual` does the same thing though. It seems to compare all key-value pairs independently regardless of ordering.